### PR TITLE
add support for key name iterator

### PR DESCRIPTION
**Pull request by manuelm**
first commit adds support for python.

### CODE REVIEW

File: src/bindings/cpp/include/key.hpp [hunk 1]
```diff
@@ -238,6 +238,8 @@ class NameIterator
 public:
 	typedef std::string value_type;
 	typedef std::string reference;
+	typedef std::string pointer;
+	typedef int difference_type;
```
Comment (manuelm) [inline src/bindings/cpp/include/key.hpp]: in case someone asks: the types are required for SWIGs iterator class for python

**Comment by manuelm** [comment]
lua support added. LuaSTLIterator templates, classes and fragments are based on ideas of SWIGs SwigPyIterator.

@markus2330 please don't merge until your Java bindings are ready so I can modify my lua bindings if necessary.

**Comment by manuelm** [comment]
jenkins build bindings please

**Comment by markus2330** [comment]
The jna binding does not leak a user visible name at all. Given the Key k:

		for (String s: k) {
			System.out.println(s);
		}

Could you please remove the WIP: and ping me when the pull request is final?

**Comment by manuelm** [comment]
nice. then it's already ready.

But can you please first merge #244 ? I'll probably need to rebase this PR afterwards

**Comment by markus2330** [comment]
Ok I merged it, please rebase then and ping me.

Could you please also add some lines in doc/NEWS.md for the release notes? Especially, the user visible changes should be described.

**Comment by manuelm** [comment]
@markus2330 news done. go ahead.

File: src/bindings/cpp/include/key.hpp (hunk 1 above)
Comment (markus2330) [inline src/bindings/cpp/include/key.hpp]: Can you document it as code-comment. Not everyone will read this here :-)

Comment (manuelm) [inline src/bindings/cpp/include/key.hpp]: grr :-)

**Comment by markus2330** [comment]
for the news: dup() and copy modules changed also, didn't they?

File: src/bindings/cpp/include/key.hpp (hunk 1 above)
Comment (markus2330) [inline src/bindings/cpp/include/key.hpp]: I have to admit I could have said it earlier ;)

**Comment by manuelm** [comment]
doc-comment done.

Can you please add news about copy module yourself? dup isn't new. Just has been fixed.
I'll take a look at the mingw build env instead.

**Comment by markus2330** [comment]
Thank you!

**Pull request merged**